### ISSUE TITLE
Suggested git pre-commit only works for single files

### ISSUE THREAD

**Issue by jestelle** [JamieMason/ImageOptim-CLI#102]
Hi there,

I tried using the suggested git pre-commit and found it only works when a single file is matched.

It also has some issues if there are spaces in the file path, with git add'ing things back in.

Cheers!

**Comment by JamieMason** [comment]
Hi @jestelle, this was changed by @MSch in [this diff](https://github.com/JamieMason/ImageOptim-CLI/commit/0c48333d57f88814c802554f8d95551d1b8cab90). 

Does the previous pre-commit hook below work for you?

```bash
git diff --cached --name-only --diff-filter=ACM | imageOptim # [options]
```

**Comment by JamieMason** [comment]
Did you try the above please @jestelle?

**Comment by jestelle** [comment]
I haven't had a chance to test it yet.  Sorry!  Will try to give it a go
soon.

**Comment by JamieMason** [comment]
No problem, thanks.

**Comment by michaelhebert** [comment]
I have the same problem. When more than 1 file is found by the git diff, $images is "file1 file2" and this triggers the error "Ignored as neither file or directory"

**Comment by JamieMason** [comment]
This hook was kindly given to use by @MSch in pull #82, maybe he can help us?

**Comment by MSch** [comment]
Sorry, I'm no shell wizard either. Is it possible to use `xargs`?

**Comment by michaelhebert** [comment]
Here's what seems to work:
```
    # run ImageOptim
    images=
    images=$(git diff --exit-code --cached --name-only --diff-filter=ACM -- $images_dir $images_dir2)
    if [ -n "$images" ]; then
        echo $images | xargs image_optim --no-pngout --no-svgo
        echo $images | xargs git add 
    fi;
```

**Comment by JamieMason** [comment]
This should be fixed in the docs for imageoptim-cli@1.14.8, thanks to https://github.com/JamieMason/ImageOptim-CLI/pull/104
